### PR TITLE
Request to add HybridNets

### PR DESCRIPTION
**Pull request by xoiga123**
Please let me know if there are any errors, thank you!

### CODE REVIEW

**Comment by xoiga123** [comment]
Seems like I'm missing some critical requirements `pretrainmodels`, `efficientnet_pytorch`. I don't think it's necessary for all the other nets in Hub, so putting them in `install.sh` is kind of bad. Should I do like YOLOv5 and use `pip install` in Hub entrypoint?

**Comment by xoiga123** [comment]
Our test is passing fine now, although some nets from `pytorch_vision` are faulting. Looking forward to your comments :)

File: datvuthanh_hybridnets.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+## Before You Start
+
+Start from a **Python>=3.7** environment with **PyTorch>=1.10** installed. To install PyTorch see [https://pytorch.org/get-started/locally/](https://pytorch.org/get-started/locally/). To install HybridNets dependencies:
+```bash
+pip install -qr https://raw.githubusercontent.com/datvuthanh/HybridNets/main/requirements.txt  # install dependencies
+```
+
+## Model Description
+ 
+<img src="https://github.com/datvuthanh/HybridNets/raw/main/images/hybridnets.jpg">
```
Comment (vmoens) [inline datvuthanh_hybridnets.md:27]: Might be a display issue from netlify or something on my side but this image appears to be quite big on https://pytorch.org/hub/, perhaps could you reshape it with `width="400"`  or something like it?

File: datvuthanh_hybridnets.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+model = torch.hub.load('datvuthanh/hybridnets', 'hybridnets', pretrained=True)
+
+#inference
+img = torch.randn(1,3,640,384)
+features, regression, classification, anchors, segmentation = model(img)
+```
+
+### Citation
+
+If you find our [paper](https://arxiv.org/abs/2203.09035) and [code](https://github.com/datvuthanh/HybridNets) useful for your research, please consider giving a star :star: and citation :pencil: :
```
Comment (vmoens) [inline datvuthanh_hybridnets.md:89]: I don't think `:star:` and `:pencil:` will appear on https://pytorch.org/hub/, though they do on github

**Review by vmoens — COMMENTED** [review]
Thanks for your contribution @xoiga123!
Can you add a dependency list in [hubconf.py](https://github.com/datvuthanh/HybridNets/blob/main/hubconf.py), something along the line of:
```
dependencies = ["albumentations", "efficientnet_pytorch", "matplotlib",
                "numpy", "opencv_python_headless", "prefetch_generator",
                "pretrainedmodels", "PyYAML", "scipy", "seaborn",
                "tensorboardX", "timm", "torch", "torchvision", "tqdm",
                "webcolors"]
```
(see https://pytorch.org/docs/stable/hub.html#how-to-implement-an-entrypoint)

File: datvuthanh_hybridnets.md (hunk 1 above)
Comment (xoiga123) [reply datvuthanh_hybridnets.md:27]: Oh sorry for not noticing that, it's on my end too. I'll try `width="800"`.

File: datvuthanh_hybridnets.md (hunk 2 above)
Comment (xoiga123) [reply datvuthanh_hybridnets.md]: Removed emojis, ty.

**Comment by xoiga123** [comment]
> Thanks for your contribution @xoiga123! Can you add a dependency list in [hubconf.py](https://github.com/datvuthanh/HybridNets/blob/main/hubconf.py), something along the line of:
> 
> ```
> dependencies = ["albumentations", "efficientnet_pytorch", "matplotlib",
>                 "numpy", "opencv_python_headless", "prefetch_generator",
>                 "pretrainedmodels", "PyYAML", "scipy", "seaborn",
>                 "tensorboardX", "timm", "torch", "torchvision", "tqdm",
>                 "webcolors"]
> ```
> 
> (see https://pytorch.org/docs/stable/hub.html#how-to-implement-an-entrypoint)

What does torch hub do with this list of dependencies though? YOLOv5 doesn't have it and YOLOP only has `torch` as a dependency. I've updated with several things to load the model (at least from `hubconf.py` onwards), but does it hurt if I included too much or too little?
```python
dependencies = ["efficientnet_pytorch",
                "pretrainedmodels", "PyYAML",
               "timm", "torch", "torchvision"]
```

**Comment by xoiga123** [comment]
@vmoens sorry to bother you, but why does the test fail with `RuntimeError: Missing dependencies: PyYAML` when I put PyYAML into `dependencies`? Should I leave it out or put it into `install.sh`? Then again, YOLOv5 imported PyYAML and passed the test without `dependencies` 😖

**Comment by vmoens** [comment]
> @vmoens sorry to bother you, but why does the test fail with `RuntimeError: Missing dependencies: PyYAML` when I put PyYAML into `dependencies`? Should I leave it out or put it into `install.sh`? Then again, YOLOv5 imported PyYAML and passed the test without `dependencies` 😖

The purpose of the `dependencies` list is for torch hub to check dependencies before importing a model, but not to install them. If yaml is not a dependency of your model I would suggest to remove it from the list (then the test should pass).
Hope that helps!

**Comment by xoiga123** [comment]
> The purpose of the `dependencies` list is for torch hub to check dependencies before importing a model, but not to install them. If yaml is not a dependency of your model I would suggest to remove it from the list (then the test should pass). Hope that helps!

I see. Can you have a final check, please? I think it's all good for now :innocent:. Future modifications should be on our repo, right?

**Comment by vmoens** [comment]
> Future modifications should be on our repo, right?

That would depend on what modifications you'd need to do 😉 but i guess that in general the answer is yes!

For the rest it looks great, thanks again for contributing with this nice piece of work!

**Pull request merged**